packages/react-checkbox/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -6,130 +6,132 @@ import { Icon } from 'office-ui-fabric-react/lib/Icon';
 
 const getClassNames = classNamesFunction<ICheckboxStyleProps, ICheckboxStyles>();
 
-export const CheckboxBase = React.forwardRef((props: ICheckboxProps, forwardedRef: React.Ref<HTMLDivElement>) => {
-  const {
-    disabled,
-    inputProps,
-    name,
-    ariaLabel,
-    ariaLabelledBy,
-    ariaDescribedBy,
-    ariaPositionInSet,
-    ariaSetSize,
-    title,
-    label,
-    checkmarkIconProps,
-    styles,
-    theme,
-    className,
-    boxSide = 'start',
-  } = props;
-
-  const id = useId('checkbox-', props.id);
-
-  const rootRef = React.useRef<HTMLDivElement | null>(null);
-  const mergedRootRefs: React.Ref<HTMLDivElement> = useMergedRefs(rootRef, forwardedRef);
-  const inputRef = React.useRef<HTMLInputElement>(null);
-
-  const [isChecked, setIsChecked] = useControllableValue(props.checked, props.defaultChecked, props.onChange);
-  const [isIndeterminate, setIsIndeterminate] = useControllableValue(props.indeterminate, props.defaultIndeterminate);
-
-  useFocusRects(rootRef);
-  useDebugWarning(props);
-  useComponentRef(props, isChecked, isIndeterminate, inputRef);
-
-  const classNames = getClassNames(styles!, {
-    theme: theme!,
-    className,
-    disabled,
-    indeterminate: isIndeterminate,
-    checked: isChecked,
-    reversed: boxSide !== 'start',
-    isUsingCustomLabelRender: !!props.onRenderLabel,
-  });
-
-  const onChange = (ev: React.ChangeEvent<HTMLElement>): void => {
-    if (isIndeterminate) {
-      // If indeterminate, clicking the checkbox *only* removes the indeterminate state (or if
-      // controlled, lets the consumer know to change it by calling onChange). It doesn't
-      // change the checked state.
-      setIsChecked(!!isChecked, ev);
-      setIsIndeterminate(false);
-    } else {
-      setIsChecked(!isChecked, ev);
-    }
-  };
-
-  const defaultLabelRenderer = React.useCallback(
-    (checkboxProps?: ICheckboxProps): JSX.Element | null => {
-      if (!checkboxProps) {
-        return null;
-      }
-      return checkboxProps.label ? (
-        <span aria-hidden="true" className={classNames.text} title={checkboxProps.title}>
-          {checkboxProps.label}
-        </span>
-      ) : null;
-    },
-    [classNames.text],
-  );
-
-  const onRenderLabel = props.onRenderLabel || defaultLabelRenderer;
-
-  const ariaChecked: React.InputHTMLAttributes<HTMLInputElement>['aria-checked'] = isIndeterminate
-    ? 'mixed'
-    : isChecked
-    ? 'true'
-    : 'false';
-
-  const slotProps = {
-    root: {
-      className: classNames.root,
-      title,
-      ref: mergedRootRefs,
-    },
-    input: {
-      className: classNames.input,
-      type: 'checkbox' as React.InputHTMLAttributes<HTMLInputElement>['type'],
-      ...inputProps,
-      ref: inputRef,
-      checked: !!isChecked,
+export const CheckboxBase: React.FunctionComponent<ICheckboxProps> = React.forwardRef<HTMLDivElement, ICheckboxProps>(
```

Comment:
I'm wondering why move the parameter typings to the generic arguments, rather than leave them on the parameters? It seems less readable this way. (Same everywhere)

Reply:
I think this may be a stylistic preference--it's more redundant with the types specified on both sides. You can still see the type if desired using intellisense.

Reply:
it's less to type when using the generic type. and I don't think there is a readability issue with either way of doing typing. Also, say if later react decides to change the typing for the `React.forwardRef` signature, using the generic types helps the such change to be less disruptive for our code.